docker/qgisserver/geomapfish_qgisserver/accesscontrol.py

```diff
@@ -728,7 +745,31 @@ def authorizedLayerAttributes(  # pylint: disable=invalid-name
             )
             return []
 
-        return attributes
+        session = self.DBSession()
+        try:
+            layers = self.get_layers(session)
+            ogc_layer_name = self.ogc_layer_name(layer)
+            if ogc_layer_name not in layers:
+                return []
+            gmf_layers = layers[ogc_layer_name]
+            protected_attributes = []
+            for gmf_layer in gmf_layers:
+                for metadata in gmf_layer.get_metadata("protectedAttribute"):
+                    protected_attributes.extend(metadata.value.split(","))
+
+            attributes_access = [a for a in attributes if a not in protected_attributes]
+            for role in roles:
+                for functionality in role.functionalities:
+                    if functionality.name == "attribute_access" and ":" in functionality.value:
+                        layer_name, layer_protected_attributes = functionality.value.split(":", 1)
+                        if layer_name == ogc_layer_name:
+                            for protected_attribute in layer_protected_attributes.split(","):
+                                if protected_attribute in attributes:
+                                    attributes_access.append(protected_attribute)
+            return attributes_access
+
+        finally:
+            session.close()
```

Comment:
A return statement in a finally?
It looks to be dangerous.

Reply:
It returns `None` in case of error, shouldn't we return explicitly an empty list, but outside the finally, or in an except.
Surprised that the linter did not complained about inconsistent return statements.

Reply:
Sorry, if there is an error, it does not return None, the error is raised.

All other python function called by C++ are wrapped in a try with an except.
I suppose QGIS may crash in some cases if such function raise an error.

Reply:
Sorry another time, other function reraise in case of error, but also log the error.
Shouldn't we do the same here for consistency ?

Reply:
+1, done